### ISSUE TITLE
x86-16bit asm.segoff config is not preserved in some cases

### ISSUE THREAD

**Issue by lasek0** [radareorg/radare2#19599]
## Environment

```
$ date
wto, 18 sty 2022, 22:08:11 CET
$ r2 -v
radare2 5.5.5 27758 @ linux-x86-64 git.5.5.4
commit: 4884287b81a3fdd1d9c81cceaccce45b1cebdf31 build: 2022-01-18__21:56:56
$ uname -ms
Linux x86_64
$ 
```

## Description

asm.segoff config is not preserved in some cases for x86-16bit, in general it is not preserved when disassembly of analysed function ("pd" or "pdf"), or after load binary file using "o" command

## Test

test1:
```
$ r2 -
e asm.arch=x86
e asm.bits=16
e asm.seggrn=8
e asm.segoff=false

wx 55c3

pd 1
e asm.segoff

af
e asm.segoff

pd 2
e asm.segoff
```

test2:
```
$ echo -ne "\x55\xc3" > test.bin
$ r2 -
e asm.arch=x86
e asm.bits=16
e asm.seggrn=8
e asm.segoff=false

e asm.segoff

o test.bin 0 rwx
e asm.segoff

e asm.segoff=false
pd 2
pd 2

af
e asm.segoff

pd 2
e asm.segoff
```

test3:
```
$ echo -ne "\x55\xc3" > test.bin
$ r2 test.bin
e asm.arch=x86
e asm.bits=16
e asm.seggrn=8
e asm.segoff=false

pd 2

pd 2

af
e asm.segoff

pd 2
e asm.segoff
```

## Output

![Screenshot from 2022-01-18 21-59-20](https://user-images.githubusercontent.com/79937932/150019824-9d379b0b-4b1d-49b2-a62c-d00e34bf5331.png)
![Screenshot from 2022-01-18 22-02-12](https://user-images.githubusercontent.com/79937932/150019833-2d84a313-5d6d-46e5-b34b-05498a082109.png)
![Screenshot from 2022-01-18 22-03-45](https://user-images.githubusercontent.com/79937932/150019848-0264e1b4-06e7-4100-b129-854453dd6a02.png)

**Comment by trufae** [comment]
is this fixed?

**Comment by lasek0** [comment]
Yes and No. Yes I have found places where this can be fixed, and No because solutions does not satisfy me. In general each time when `r_config_set_i` is called with "asm.bits" = 16 then segoff is set. I think that asm.segoff should be only set when e sam.bits = 16 is executed from console or from script file, but not as direct call via `r_config_set_i`. That is why I not pushed to PR yet for that issue. And  quick simple restore segoff for pd is not valid way to go. I fixed this by save and restore around `r_config_set_i ("asm.bits", ...)` in few places. But as I said this can be done better.

**Comment by trufae** [comment]
Got it, thanks for the explanation

**Comment by trufae** [comment]
i thikn the best way to proceed is to add some tests for this, can you do that? @lasek0

**Comment by trufae** [comment]
Ping?
